### DISCUSSION TITLE
2FA without mobile

### DISCUSSION THREAD

**Discussion by jonlepage** [community/community#140230] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello, I have a problem.
I do not have a mobile phone because there is no phone coverage in my area.
I only have a good old landline.

2FA makes me very anxious because you sent me an email saying that I will lose my access without being able to do two-factor authentication with a mobile device.
A mobile for me it a very dangerous thing !
i prefer keep a computer and when i have to go in the city i dont bring phone for my security.

How can I bypass the 2FA, given that I am in a private lab and no one has access to my encrypted computer systems?

I ask this because i will lose github service? if i not buy a mobile and subscribing to a ~$200 satellite monthly plan for mobile support in my lost areas in Canada ?

![image](https://github.com/user-attachments/assets/00e2dfc8-abd8-4fb9-a870-b59ca2520bf0)

Thank you.

**Comment by feijoes** [comment]
I found this video explaining how to setup 2FA in desktop apps (KeePass and Safe in Cloud):
https://www.youtube.com/watch?v=ib1hpFWMW6w
(note it's from 2017)

**Reply by jonlepage** [reply]
> I found this video explaining how to setup 2FA in desktop apps (KeePass and Safe in Cloud): https://www.youtube.com/watch?v=ib1hpFWMW6w (note it's from 2017)

hi, thank you for your time, I understand your good intentions, but when I work, I need to stay focused.
**I can't afford to install third-party software to manage my security this way.**

I already have very efficient software, and all my passwords are managed with https://www.lastpass.com/.
All my passwords look like this, including the master password written on a sheet: `4h@UosYZFU9^1sdOGQ#@1swM2XPF9ftq`.
In theory, **unless a major breakthrough in quantum computing occurs in the coming years**, it would take ~thousands of years to brute force such a password.
**This is the only optimal way to secure your digital environment.**

**Adding a mobile device or even using a mobile device is only useful for telemetry for big tech compagnie and has no security benefit.**

The solution would be for **GitHub to ask for my driver's license or proof of identity**, which I find more secure than linking dangerous hardware, which is impossible to use in my region anyway.
Therefore, I am looking for a more productive and flexible solution, knowing that I pay for GitHub's lower-tier services.
It's simply a matter of respect for the customers who have trusted these services and allowed GitHub to be what it is today.
I might make the effort if I at least had satellite coverage, but that costs $200/m.
So I only have internet and telephone lines.


I think it’s pointless to remind what happens to Goliath when he believes he can do whatever he wants, as everyone is now dependent on his services and centralization.

![image](https://github.com/user-attachments/assets/2a4f2418-6cd0-404e-af36-8d042c31535b)
![image](https://github.com/user-attachments/assets/ad5db3ba-2539-4e00-bb48-4324c431782e)

Do not impose directives on your clients by lying about your true intentions.

**Reply by feijoes** [reply]
I understand your situation, but I believe you're at a bit of a dead end. You could try contacting GitHub Support or setting up 2FA on a device initially and then using recovery codes. GitHub provides these one-time-use codes when you set up 2FA, which you can use these codes to regain access. This way, you won’t need to pay for a "$200 satellite monthly plan", just a one-time setup.

**Reply by jonlepage** [reply]
> e

I thought about using this, but some told me it was dangerous and not good idea.
https://beeceptor.com/resources/2fa-code-generator/

So i am compromising my safety for their new security service.
It makes no sense.
How i can generate easly 2fa code for free with a online service ?
I didn't need all this fog this week !
Right now, they’re making my life difficult and change my habit from ~20ry , forcing me to do things I don’t understand and look risky, and it’s going to take me many weeks to study and understand what they are doing !

**Comment by iamasadshah** [comment]
Hey, I totally get where you're coming from. It must be frustrating dealing with 2FA when you don’t have access to a mobile phone, especially given your unique situation. In my opinion, there’s no need to stress over buying a phone or paying for an expensive satellite plan just to access GitHub.
I think there are a few ways to work around it safely.

**1. TOTP with a Desktop App:**
I suggest using a desktop-based authenticator instead of a mobile one. There are great apps like Authy, 1Password, or WinAuth that you can install directly on your computer. In my experience, they work just as well as mobile apps, and once they’re set up, you don’t need an internet connection to use them. You’ll be able to get your 2FA codes from your computer without having to worry about a mobile device.

In my opinion, this is the easiest and most practical way to handle 2FA without needing a phone.

**2. Use a Security Key (FIDO U2F):**
If you want something more physical, I’d recommend looking into a security key like YubiKey. I’ve used these before, and they’re super secure and easy to use. You just plug it into your computer’s USB port when you log in to GitHub, and it acts as your second factor of authentication. No phone is required at all.

From my experience, security keys are a great alternative for those who prioritize privacy and security, like yourself.

**3. Check if You Can Disable 2FA:**
In some cases, GitHub doesn’t require 2FA depending on the type of account you have. I suggest checking your account settings to see if you can opt out of 2FA for now. If it’s not mandatory, this could save you the hassle altogether.

This is a good option if your GitHub usage is private and in a secure environment like your lab.

**4. Use Recovery Codes:**
Another thing I’d recommend is to save the recovery codes that GitHub provides when setting up 2FA. Since your computer systems are encrypted and secure, these codes could be a solid backup option. You can use them to access your account if you lose access to your 2FA method, and they don’t require a mobile device.

**Reply by jonlepage** [reply]
thanks, but they are all mobile app or paid service 
![image](https://github.com/user-attachments/assets/ad78a3e6-2036-46a5-94b7-50583c2f470f)
![image](https://github.com/user-attachments/assets/979d36c4-6b5a-45ff-b324-7e5122d4ddff)

i found this and work fine and it easy.
Compatible with chromium browser.
https://github.com/Authenticator-Extension/Authenticator

**Comment by petzval** [comment]
See my recent post for a simple method that does not require a phone or installed software or browser extension.

https://github.com/orgs/community/discussions/140205

**Reply by jonlepage** [reply, marked as answer]
> See my recent post for a simple method that does not require a phone or installed software or browser extension.
> 
> https://github.com/orgs/community/discussions/140205

i find a alternative easy to use and FREE , what a hell
https://github.com/Authenticator-Extension/Authenticator

I have never felt so vulnerable and anxious in my 20-year career !
They are holding my depot in hostage to force me to install third-party software, under threat of losing my account !
Instead of suggesting and letting people manage their own security, they preferred to impose their way of doing things.
I will remember this for a long time when it comes to trusting Microsoft again !
